### PR TITLE
feature/AKD-117: Added Azure 'NSG Flow Logs Retention Period' plugin and test cases

### PR DESCRIPTION
**Pull request by ali-imran7**

### CODE REVIEW

File: plugins/azure/networkwatchers/nsgFlowLogsRetentionPeriod.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                for (const flowLog of flowLogs.data) {
+                    let retentionDays = 0;
+                    if (flowLog.retentionPolicy &&
+                        flowLog.retentionPolicy.days) {
+                        retentionDays = flowLog.retentionPolicy.days;
+                    }
+
+                    if (retentionDays >= config.retentionDays) {
+                        helpers.addResult(results, 0,
+                            `NSG FLow Logs Data retention policy is set to ${retentionDays} of ${config.retentionDays} days desired limit`,
```
Comment (AkhtarAmir) [inline plugins/azure/networkwatchers/nsgFlowLogsRetentionPeriod.js:71]:
```suggestion
                            `NSG fLow log has retention period set to ${retentionDays} of ${config.retentionDays} days desired limit`,
```

File: plugins/azure/networkwatchers/nsgFlowLogsRetentionPeriod.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                            location, flowLog.id);
+                    } else {
+                        helpers.addResult(results, 2,
+                            `NSG FLow Logs Data retention policy is set to ${retentionDays} of ${config.retentionDays} days desired limit`,
```
Comment (AkhtarAmir) [inline plugins/azure/networkwatchers/nsgFlowLogsRetentionPeriod.js:75]:
```suggestion
                            `NSG flow log has retention period set to ${retentionDays} of ${config.retentionDays} days desired limit`,
```

File: plugins/azure/networkwatchers/nsgFlowLogsRetentionPeriod.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure/');
+
+module.exports = {
+    title: 'NSG Flow Logs Retention Period',
+    category: 'Network Watcher',
+    description: 'Ensures that Azure Network Security Groups (NSGs) have a sufficient flow log retention period',
+    more_info: 'A flow log data retention period of 90 days or more, allows you to collect the necessary amount of logging data required to check for anomalies and provide details about any potential security breach.',
+    recommended_action: 'Ensure that Network Security Groups (NSG) flow logs retention period is 90 days or more',
```
Comment (AkhtarAmir) [inline plugins/azure/networkwatchers/nsgFlowLogsRetentionPeriod.js:9]:
```suggestion
    recommended_action: 'Modify NSG flow logs and set desired value in days for retention period',
```

Comment (AkhtarAmir) [reply plugins/azure/networkwatchers/nsgFlowLogsRetentionPeriod.js:9]: I find it better to give them bit more clarity about recommended action

File: plugins/azure/networkwatchers/nsgFlowLogsRetentionPeriod.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                    helpers.addResult(results, 0, 'No flow logs data found', location);
+                    return scb();
+                }
+
+                for (const flowLog of flowLogs.data) {
+                    let retentionDays = 0;
+                    if (flowLog.retentionPolicy &&
+                        flowLog.retentionPolicy.days) {
+                        retentionDays = flowLog.retentionPolicy.days;
+                    }
```
Comment (AkhtarAmir) [inline plugins/azure/networkwatchers/nsgFlowLogsRetentionPeriod.js:67]:
```suggestion
                    if (flowLog.retentionPolicy && flowLog.retentionPolicy.days) retentionDays = flowLog.retentionPolicy.days;
```

File: plugins/azure/networkwatchers/nsgFlowLogsRetentionPeriod.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+                    return scb();
+                }
+
+                if (!flowLogs.data.length) {
+                    helpers.addResult(results, 0, 'No flow logs data found', location);
+                    return scb();
+                }
+
+                for (const flowLog of flowLogs.data) {
+                    let retentionDays = 0;
```
Comment (AkhtarAmir) [inline plugins/azure/networkwatchers/nsgFlowLogsRetentionPeriod.js:63]: Just to make sure that code doesn't break
```suggestion
                for (const flowLog of flowLogs.data) {
                    if (!flowLog.id) continue;
                    let retentionDays = 0;
```

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: collectors/azure/collector.js [hunk 6]
```diff
@@ -201,6 +201,13 @@ var postcalls = {
             url: 'https://management.azure.com/{id}/securityAlertPolicies?api-version=2017-03-01-preview'
         }
     },
+    flowLogs: {
+        listAll: {
```
Comment (AkhtarAmir) [inline collectors/azure/collector.js:205]:
```suggestion
        list: {
```

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

**Review by AkhtarAmir — APPROVED** [review]

**Review by giorod3 — APPROVED** [review]

**Pull request merged**